expression/distsql_builtin.go

```diff
@@ -43,11 +43,21 @@ func PbTypeToFieldType(tp *tipb.FieldType) *types.FieldType {
 }
 
 func getSignatureByPB(ctx sessionctx.Context, sigCode tipb.ScalarFuncSig, tp *tipb.FieldType, args []Expression) (f builtinFunc, e error) {
+	if ctx == nil {
+		panic("ctx should not be nil")
```

Comment:
Although there are `panic` codes already in TiDB, it is better to avoid it and return an `error`, since for the end-users, `panic` will lead to a closed connection rather than a simple `error` message.

Reply:
sure, but what error should we return, I "panic" this because the original implement will `panic` in `newBaseBuiltinFunc` if `ctx` is `nil`

Reply:
I think we can remove this check.